### PR TITLE
Modeling - Score v0

### PR DESCRIPTION
**Pull request by miltminz**
This PR aims at adding new features for the modelling of the pyro-score
- A new module in `datasets` called `model_dataset.py` has been added: this deals with the aggregations (by day and department) and merge of ERA5, FWI and VIIRS datasets for creating a new dataset to be used in the modelling phase
- New folder `models` with `score_v0.py`: here we add functions to process the dataset for modelling (train/test split, lags, filtering correlated features to target, random forest and XGBoost classifier)
- A script `example_scorev0.py` serving as an example of how to use the new features
- Unit tests for the new features

### CODE REVIEW

File: pyro_risks/datasets/model_dataset.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+from pyro_risks.datasets import NASAFIRMS_VIIRS, ERA5Land
+from pyro_risks.datasets.utils import get_intersection_range
+from pyro_risks.datasets.fwi import GwisFwi
+
+import pandas as pd
+
+__all__ = ["ModelDF"]
+
+
+class ModelDF(pd.DataFrame):
```
Comment (frgfm) [inline pyro_risks/datasets/model_dataset.py:10]: Perhaps we can find a more informative class name (and module name) :)

File: test/test_models.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_frame_equal, assert_series_equal
+from sklearn.datasets import load_breast_cancer
+
+from pyro_risks.models import score_v0
+
+
+class UtilsTester(unittest.TestCase):
```
Comment (frgfm) [inline test/test_models.py:11]: `ModelsTester` then ;)

File: test/test_models.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            {'day': ['2019-07-01', '2019-08-02', '2019-06-12'],
+             'departement': ['Aisne', 'Cantal', 'Savoie'],
+             'fires': [0, 5, 10],
+             'fwi_mean': [13.3, 0.9, 2.5],
+             'ffmc_max': [23, 45.3, 109.]
+             })
+        X, y = score_v0.prepare_dataset(df, selected_dep=['Aisne', 'Cantal'])
+        assert_frame_equal(
+            X, pd.DataFrame({'ffmc_max': {0: 23.0, 1: 45.3}, 'fwi_mean': {0: 13.3, 1: 0.9}}))
+        assert_series_equal(y, pd.Series([0, 1], name='classif_target'))
```
Comment (frgfm) [inline test/test_models.py:23]: perhaps there is a ̀frame_equal` boolean function in pandas we could use? (To keep the `self.assertTrue`)
Same for following tests

**Review by frgfm — COMMENTED** [review]
Thanks for the PR!
I added a few comments but it seems good to me! The unittest seems to be failing because of RAM just like earlier unittest with the full DF. Not sure what would be the best move here :man_shrugging:

**Comment by chloeskt** [comment]
To add on what @frgfm said regarding the failed tests. It is indeed due to the RAM, I believe that we need to add "mock" datasets in the release (such as what I did for ERA5, FIRMS and VIIRS here https://github.com/pyronear/pyro-risks/releases/tag/v0.1.0-data). There are two datasets tests for which this has yet to be implemented: 
- FWI (the largest one)
- NOAA 

Do you guys think that it is a good idea to proceed in such way ? @miltminz @frgfm 

In that case, I believe that you'll need to add mock test datasets for FWI Milton, and for NOAA if you can (otherwise I'll do it in another PR)

If you have better ideas, I'm all ears !

File: pyro_risks/datasets/model_dataset.py (hunk 1 above)
Comment (miltminz) [reply pyro_risks/datasets/model_dataset.py:10]: alright, I'm open to suggestions

File: test/test_models.py (hunk 3 above)
Comment (miltminz) [reply test/test_models.py:23]: what is your point? `assert_frame_equal`and `assert_series_equal` are pandas testing functions which do the job...

File: pyro_risks/datasets/model_dataset.py (hunk 1 above)
Comment (chloeskt) [reply pyro_risks/datasets/model_dataset.py:10]: Perhaps something along these lines: 
- prepData ? 
- modelingData ?

Comment (frgfm) [reply pyro_risks/datasets/model_dataset.py:10]: `WildfireTimeSeries` would be a bit general, but it's a start?
Or what about "ERAVIIRS"?

**Comment by frgfm** [comment]
@chloeskt I guess the problem will come back every time we get a big dataset.
Perhaps we could make the test only locally? It cannot actually be done on github and it sounds heavy to make a mock dataset for each set we want to upload :/

I'm fine with either way, whichever you think is best

**Comment by chloeskt** [comment]
I don't think that doing the tests locally is a good idea, plus we are going to face the same issue: for instance if your computer has only 4GB or 8GB of RAM, tests might fail. So my guess is that we need to add mock datasets, even if it's heavy and uncomfortable. But I have less experience than you guys so if it's easier/better for everyone to just run the tests locally, then fine by me !

File: pyro_risks/datasets/model_dataset.py (hunk 1 above)
Comment (miltminz) [reply pyro_risks/datasets/model_dataset.py]: done with `MergedEraFwiViirs`, I hope you like it ;-)

File: test/test_models.py (hunk 2 above)
Comment (miltminz) [reply test/test_models.py]: done

File: test/test_models.py (hunk 3 above)
Comment (frgfm) [reply test/test_models.py:23]: that the method could be a function as is, but it's really just a detail :ok_hand:

**Review by frgfm — APPROVED** [review]
Thanks for the edits!

**Review by chloeskt — APPROVED** [review]

**Pull request merged**